espnet/nets/pytorch_backend/ctc.py

```diff
@@ -80,6 +83,31 @@ def loss_fn(self, th_pred, th_target, th_ilen, th_olen):
         else:
             raise NotImplementedError
 
+    def ctc_alignment_targets(self, hs_pad, hlens, ys_pad):
+        with torch.no_grad():
+            ys = [y[y != self.ignore_id] for y in ys_pad]  # parse padded ys
+    
+            hlens = torch.from_numpy(np.fromiter(hlens, dtype=np.int64))
+            olens = torch.from_numpy(np.fromiter(
+                (x.size(0) for x in ys), dtype=np.int64))
+            ys_hat = self.ctc_lo(F.dropout(hs_pad, p=self.dropout_rate)).transpose(0, 1).log_softmax(dim=-1)
```

Comment:
`F.dropout()` needs to have an optional parameter named `self.training` to distinguish the training and inference phase. Does the `ctc_alignment_targets` function only be used in the training phase?
Related PR: https://github.com/espnet/espnet/pull/3539

Reply:
This is only used in the training to get the force alignment from CTC.

Reply:
But you are right this is something interesting to consider